replicate/snapshot.go

```diff
@@ -103,17 +104,29 @@ func (sess *SnapshotReplicateSession) Run() error {
 		return errors.Trace(err)
 	}
 	metrics.AddCounter(metrics.SnapshotTotalSizeCounter, float64(snapshotFileSize), tableFQN)
+	errCh := make(chan error, 1)
+	var wg sync.WaitGroup
 	if err := sess.externalStorage.WalkDir(sess.ctx, opt, func(path string, size int64) error {
 		if strings.HasSuffix(path, CSVFileExtension) {
-			if err := sess.loadSnapshotDataIntoDataWarehouse(path); err != nil {
-				return errors.Annotate(err, "Failed to load snapshot data into data warehouse")
-			}
-			metrics.AddCounter(metrics.SnapshotLoadedSizeCounter, float64(size), tableFQN)
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				if err := sess.loadSnapshotDataIntoDataWarehouse(path); err != nil {
+					sess.logger.Error("Failed to load snapshot data into data warehouse", zap.Error(err), zap.String("path", path))
+					errCh <- errors.Annotate(err, "Failed to load snapshot data into data warehouse")
```

Comment:
Looks like errors channel may be full thus blocking this part?

Reply:
fixed in https://github.com/pingcap-inc/tidb2dw/pull/74/commits/c8d985c94247a25488c8474df8e2efab67a5144f, set channel size to `fileCount`.